fancy-weather/src/index.js

```diff
@@ -0,0 +1,426 @@
+      data.weather.scale = 'celsium';
+      data.weather.currentDescription = jsonResponse.current.weather[0].description;
+      data.weather.currentFeelsLike = jsonResponse.current.feels_like;
+      data.weather.currentWindSpeed = jsonResponse.current.wind_speed;
+      data.weather.currentHumidity = jsonResponse.current.humidity;
+      data.weather.currentIcon = jsonResponse.current.weather[0].icon;
+      data.weather.nextFirstDayTemp = jsonResponse.daily[1].temp.day;
+      data.weather.nextFirstDayIcon = jsonResponse.daily[1].weather[0].icon;
+      data.weather.nextSecondDayTemp = jsonResponse.daily[2].temp.day;
+      data.weather.nextSecondDayIcon = jsonResponse.daily[2].weather[0].icon;
+      data.weather.nextThirdDayTemp = jsonResponse.daily[3].temp.day;
+      data.weather.nextThirdDayIcon = jsonResponse.daily[3].weather[0].icon;
+      return data;
+    });
+}
+
+function determineTimeOfDay() {
+  switch (true) {
+    case data.time.hour >= 5 && data.time.hour <= 11:
+      timeOfDay = 'morning';
+      break;
+    case data.time.hour >= 18 && data.time.hour <= 21:
+      timeOfDay = 'evening';
+      break;
+    case (data.time.hour > 21 && data.time.hour <= 24) || (data.time.hour >= 0 && data.time.hour < 5):
+      timeOfDay = 'night';
+      break;
+    case data.time.hour > 11 && data.time.hour < 18:
+      timeOfDay = 'day';
+      break;
+    default:
+      timeOfDay = '';
+  }
+}
+
+function determineTheSeason() {
+  switch (data.time.month) {
+    case 11:
+    case 0:
+    case 1:
+      timeOfTheYear = 'winter';
+      break;
+    case 2:
+    case 3:
+    case 4:
+      timeOfTheYear = 'spring';
+      break;
+    case 5:
+    case 6:
+    case 7:
+      timeOfTheYear = 'summer';
+      break;
+    case 8:
+    case 9:
+    case 10:
+      timeOfTheYear = 'autumn';
+      break;
+    default:
+      timeOfTheYear = '';
+  }
+}
+
+function getRandomNumber(min = 0, max = 100) {
+  const num = Math.floor(min + Math.random() * (max + 1 - min));
+  return num;
+}
+
+function loadPictures() {
+  determineTimeOfDay();
+  determineTheSeason();
+  console.log(timeOfDay, timeOfTheYear);
+  return fetch(
+    `https://www.flickr.com/services/rest/?method=flickr.photos.search&api_key=c5147c8dda00e6d3b6010a9f73970df1&tags=weather,nature,${timeOfDay},${timeOfTheYear},-people,-women,-girl,-child,-children,-men,-religion&tag_mode=all&extras=url_h&format=json&nojsoncallback=1&per_page=100&media=photos&sort=relevance&content_type=1`,
+  )
+    .then((response) => response.json())
+    .then((jsonResponse) => {
+      console.log(jsonResponse);
+      let randomNumber = getRandomNumber(0, jsonResponse.photos.photo.length - 1);
+      if (jsonResponse.photos.photo[randomNumber].url_h === undefined) {
+        do {
+          console.log('Перезапуск без фотки');
+          randomNumber = getRandomNumber(0, jsonResponse.photos.photo.length - 1);
+        } while (!jsonResponse.photos.photo[randomNumber].url_h);
```

Comment:
In some causes it can be infinite cycle. So, maybe it's better to have some condition to stop cycle